### ISSUE TITLE
[Workspace Feeds] Create a new form for adding missing address and personal information data to ship a card

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#48487]
### Problem

When admin issues a physical card to an employee who does not have all the necessary personal details set up that are required to send the card to them, we create this card in NOT_ISSUED state and we ask the member to add the missing info. In oldDot, this is done by inbox task. We need the personal address, phone number and legal first and last name to be able to ship the card.

In Newdot, however, there is no specific form for this information altogether, leading to a schizophrenic experience that is not clear to the customer. They need to add the address and legal name in 2 different forms in settings at the moment and the only way to add a phone number is to add a secondary login which is another form.

### Solution

When a member needs to add the missing details, this report action shows to them. 

<img width="1713" alt="image" src="https://github.com/user-attachments/assets/e7ce9956-8aa9-4794-9b91-3a087866d2c6">

Let's create a new form that will open in RHP and will collect all the data we need to ship the card. This would then call the existing `Card_SetUserDetails` API command, which requires these fields:

```
'legalFirstName' => 'First name',
            'legalLastName' => 'Last name',
            'phoneNumber' => 'Phone number',
            'addressCountry' => 'Country',
            'addressCity' => 'City',
            'addressStreet' => 'Street',
            'addressZip' => 'Zip code',
```

The API params are authToken and `data`, which is a stringified object with the aforementioned properties

There is also a change required in the backend to make sure the `Card_SetUserDetails` calls the `ShipExpensifyCard` Auth command if there are already some cards in the not_issued state for cards.

**Comment by dubielzyk-expensify** [comment]
Here's an idea of how we could solve it. I used the page header with the progress bar stepper at the top and made each information it's own screen. That way we can remove screens where we have the information. Or if we want, we can pre-fill it.

Also, I'm assuming the phone number needs validation? Let me know if I've gotten any of these screens or information wrong.

![CleanShot 2024-09-04 at 16 00 26@2x](https://github.com/user-attachments/assets/9433af64-29e2-4c29-9161-b844dfae7140)

If we wanna use our regular stepper it would look like:
![CleanShot 2024-09-04 at 16 03 58@2x](https://github.com/user-attachments/assets/a9db792f-5973-4a00-8265-9dce08a5f0d0)

cc @Expensify/design

**Comment by shawnborton** [comment]
Nice! The second set using the regular stepper feels more natural to me given that we use this a lot in other places of the app. I also like the idea of pre-filling the fields where we can.

**Comment by MariaHCD** [comment]
The second set also looks great to me 👍🏼 

Just a note about `ShipExpensifyCard`, it currently requires a cardID as a param. I think we can have the FE pass the cardID to `Card_SetUserDetails` - perhaps we can look for the first NOT_ISSUED card in the user's card list.

**Comment by dannymcclain** [comment]
I love the look of the first set but agree the second set feels more at home in the product right now. 

**Nitpick incoming:** What's your gap between the magic code fields and the `Didn't receive a magic code text?`? Feels a little tight to me. I've been using `20px` in the mocks for various other flows we've been validating recently. Feel free to ignore!

![image](https://github.com/user-attachments/assets/6d12ccf0-0bd1-46cc-9f32-7ff24c786c79)

**Comment by mountiny** [comment]
@dubielzyk-expensify Thanks! That looks great.

I am not sure if we need the validation, we do not need it for the card creation or the 3ds purposes. So I think we can leave it out. We are not going to create secondary logins using the phone number for now so we do not need validation.

**Comment by mountiny** [comment]
![image](https://github.com/user-attachments/assets/24e6c7e5-e3c2-4a3b-8566-ad643ba4beaf)

In newDot we are also asking for date of birth, but I dont think its actually needed

**Comment by mountiny** [comment]
@madmax330 noted that this is not required for marqeta so we are most likely fine to leave this out from the NewDot form https://expensify.slack.com/archives/C036QM0SLJK/p1725463473566129?thread_ts=1725283960.380699&cid=C036QM0SLJK

**Comment by shawnborton** [comment]
I dig it!

**Comment by mountiny** [comment]
Discussing if @allgandalf or @DylanDylann can start working on this now

**Comment by mountiny** [comment]
I have just been working on the backend code and the date of birth is ALSO required for marqeta. [internal link](https://github.com/Expensify/Auth/blob/812a466ef42d34fb4b7335bc0115c7941a9ec18a/auth/lib/Account.cpp#L99-L107)

@dubielzyk-expensify @shubham1206agra we need to add that into the form

**Comment by shawnborton** [comment]
Agree, I think I like it more as a separate step like you have in the first series there.

**Comment by trjExpensify** [comment]
👋 Discussed this in the wave-collect thread [here](https://expensify.slack.com/archives/C036QM0SLJK/p1725971436244989?thread_ts=1725898727.642509&cid=C036QM0SLJK), and made a few changes to those mocks. CC: @kevinksullivan @mountiny 

![image](https://github.com/user-attachments/assets/6951c9e7-3de1-4a27-b5c7-abe963102a4b)

Summary of changes: 

- Changed the button in the system message from `Add person details` to `Add shipping details`.
- updated the system message text copy to reflect the prompt to add shipping details: "issued @cardholder an Expensify Card! The card will be shipped once shipping details are added."
- update the system message text to use supporting text font, as it’s a system message and the mocks were in text font.
- updated the page header titles to `Add shipping details` to match the button name
- Included a mock for the system message copy when the shipping details are successfully added for completeness, and updated that copy to reflect shipping details: "added shipping details. Expensify Card will arrive in 2-3 business days."
- Changed the title copy of the phone number screen to `Enter phone number` so it matches all of the other screens which say “Enter xx”. I didn't see a good reason why that one would be different to the other steps and say "Add phone number". 

@shubham1206agra can you get that updated in the PR, please? Once those are in, we can obtain confirmation on the proposed Spanish translations.

**Comment by mountiny** [comment]
Now we have the base of the form in the App, and users can use it to ship the card; thank you, @shubham1206agra! As discussed, though, let's improve the form further by:
- giving each step a separate URL so the navigation patterns work across all the platforms
- using onyx form drafts so the state of the form is not lost when user refreshes the modal
- Add the confirmation page to the end of the flow before we submit

![image](https://github.com/user-attachments/assets/ea9676d3-f955-4162-b6e8-7f33e9c20a72)

**Comment by mountiny** [comment]
This flow is not trivial so I think $750 total price for @shubham1206agra implementing these flows is fair reward once the changes from comment above are delivered.

**Comment by mountiny** [comment]
@shubham1206agra, how is it looking on the follow-ups? can you please prioritize them over taking on new work? Thanks

**Comment by mountiny** [comment]
> I may get away with using same URL for the entire flow.

I think we should use a different url, otherwise, the browser history wont work across all platforms and reliably/ predictably/ without adding some hacks.

the bank account flow of submit expense forms have all its own url for each step

**Comment by alexpensify** [comment]
@shubham1206agra any update here? Thanks!

**Comment by alexpensify** [comment]
Alright, we are waiting for this one to go into production

**Comment by mountiny** [comment]
Seems like it is, assigned yall to help resolve it

**Comment by mountiny** [comment]
I believe this is now ready for payment, I believe from the previous comment, the price was $750

**Comment by alexpensify** [comment]
@shubham1206agra and @allgandalf, please apply through the Upwork Job link, and then I can complete the payment process. Thanks! 

https://github.com/Expensify/App/issues/48487#issuecomment-2407896324

**Comment by alexpensify** [comment]
Offers have been sent via Upwork, please accept and I can complete the payment process. Thanks!

**Comment by alexpensify** [comment]
Closing - everyone has been paid via Upwork.
